### PR TITLE
reorganize code and add an unit test

### PR DESCRIPTION
**Pull request by salvete**
**What this PR does / why we need it**:

 Reorganize the code to make the structure clearer. And also add an unit test for the internal ErofsCache.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: src/overlaybd/tar/erofs/erofs_fs.cpp [hunk 1]
```diff
@@ -21,11 +21,30 @@
 #include <dirent.h>
 #include <vector>
 
+struct erofs_fs_private {
+	struct erofs_sb_info sbi;
+	struct liberofs_file target_file;
+	unsigned long magic = LIBEROFS_MAGIC;
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/erofs_fs.cpp:27]: what's the exact use of this fields?

File: src/overlaybd/tar/erofs/erofs_fs.h [hunk 2]
```diff
@@ -21,13 +21,10 @@
...
 #include <photon/common/alog.h>
 #include <vector>
-#include "erofs_common.h"
-#include "erofs/internal.h"
 
 class ErofsFileSystem: public photon::fs::IFileSystem {
 public:
-	struct erofs_sb_info sbi;
-	struct liberofs_file target_file;
+	void *fs_private;
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/erofs_fs.h:27]: I think you could just
```
struct ErofsFileSystemInt;
ErofsFileSystemInt *private;
```

File: src/overlaybd/tar/erofs/erofs_fs.cpp (hunk 1 above)
Comment (hsiangkao) [reply src/overlaybd/tar/erofs/erofs_fs.cpp:27]: I mean what's the uses of `magic`?

Comment (salvete) [reply src/overlaybd/tar/erofs/erofs_fs.cpp:27]: Just a double check for pointer conversion using reinterpret_cast.

File: src/overlaybd/tar/erofs/erofs_fs.h [hunk 3]
```diff
@@ -61,7 +58,7 @@ class ErofsFileSystem: public photon::fs::IFileSystem {
 class ErofsFile: public photon::fs::VirtualReadOnlyFile {
 public:
 	ErofsFileSystem *fs;
-	struct erofs_inode inode;
+	void *file_private;
```
[the remark this answers is not in the document]
Comment (salvete) [reply src/overlaybd/tar/erofs/erofs_fs.h:61]: OK, I will give it a try.

File: src/overlaybd/tar/erofs/erofs_fs.h (hunk 2 above)
Comment (salvete) [reply src/overlaybd/tar/erofs/erofs_fs.h:27]: Done.

**Comment by BigVan** [comment]
LGTM

**Pull request merged**